pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,171 @@
+		t.Fatalf("Valid config should not return an error, got: %v", diagnostics)
+	}
+
+	ret, err := component.RenderManifests()
+	if err != nil {
+		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
+	}
+
+	return ret
+}
+
+//nolint:funlen
+func TestConversion(t *testing.T) {
+	testCases := []struct {
+		Name                   string
+		InputConfig            string
+		ExpectedConfigFileName string
+		Expected               reflect.Value
+		JSONPath               string
+	}{
+		{
+			Name: "use external_url param",
+			InputConfig: `
+component "prometheus-operator" {
+  prometheus {
+    external_url = "https://prometheus.externalurl.net"
+  }
+}
+`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			Expected:               reflect.ValueOf("https://prometheus.externalurl.net"),
+			JSONPath:               "{.spec.externalUrl}",
+		},
+		{
+			Name: "no external_url param",
+			InputConfig: `
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			Expected:               reflect.ValueOf("https://prometheus.mydomain.net"),
+			JSONPath:               "{.spec.externalUrl}",
+		},
+		{
+			Name: "ingress creation for prometheus",
+			InputConfig: `
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/ingress.yaml",
+			Expected:               reflect.ValueOf("prometheus.mydomain.net"),
+			JSONPath:               "{.spec.rules[0].host}",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			m := renderManifests(t, tc.InputConfig)
+			if len(m) == 0 {
+				t.Fatalf("Rendered manifests shouldn't be empty")
+			}
+
+			gotConfig, ok := m[tc.ExpectedConfigFileName]
+			if !ok {
+				t.Fatalf("Config not found with filename: %q", tc.ExpectedConfigFileName)
+			}
+
+			u := getUnstructredObj(t, gotConfig)
+			got := getValFromObject(t, tc.JSONPath, u)
+
+			switch got.Kind() { //nolint:exhaustive
+			case reflect.Interface:
+				switch gotVal := got.Interface().(type) { //nolint:gocritic
+				case string:
+					expVal, ok := tc.Expected.Interface().(string)
+					if !ok {
+						t.Fatalf("expected value is not string")
+					}
+
+					if gotVal != expVal {
+						t.Fatalf("expected: %s, got: %s", expVal, gotVal)
+					}
+
+					// Add more cases as the expected values become heterogeneous.
+					// case bool:
```

Comment:
I'd put that on different indent level, maybe above `case string` ?

Reply:
Yeah, it got automatically indented forward.